mapper/cumulocity/c8y_translator_lib/src/serializer.rs

```diff
@@ -66,43 +72,22 @@ impl C8yJsonSerializer {
 
         assert!(self.timestamp_present);
 
-        self.buffer.push('}');
+        self.json.write_close_obj();
         Ok(())
     }
 
     pub fn bytes(mut self) -> Result<Vec<u8>, C8yJsonSerializationError> {
         self.end()?;
-        Ok(self.buffer.into())
-    }
-
-    fn write_key(&mut self, key: &str) {
-        self.write_str(key);
-        self.buffer.push(':');
-    }
-
-    fn write_str(&mut self, s: &str) {
-        self.buffer.push('"');
-        self.buffer.push_str(s);
-        self.buffer.push('"');
-    }
-
-    fn write_f64(&mut self, value: f64) -> std::fmt::Result {
-        use std::fmt::Write;
-        self.buffer.write_fmt(format_args!("{}", value))
+        Ok(self.json.into_string().into())
```

Comment:
It would be better to call here the function `C8yJsonSerializer::into_string()`.

Reply:
yes!

Reply:
so rename `bytes` to `into_string`.